feat: include the commit deployed in PR comments

RyanGWU82 opened the pull request:
Based on a suggestion from Kanad, I've updated PR messages to add the commit that was deployed. If you do multiple `git push`es in succession, and it takes a while for the app to redeploy, it's sometimes hard to tell exactly which commit is currently on Heroku.

Example of the new message:

<img width="925" alt="Screen Shot 2022-05-17 at 2 18 15 PM" src="https://user-images.githubusercontent.com/313895/168911935-b89d2679-f226-46a1-9270-5d351fe30a98.png">

kanadgupta commented on src/git.js:
```diff
@@ -30,6 +31,28 @@ module.exports.refExists = ref => {
 };
 
+/* Returns the SHA-1 hash of the latest commit in the given ref. */
+module.exports.shaForRef = ref => {
```
FYI most of these are available as env variables in GitHub Actions runners: https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

But if this approach appears to work just fine 🤷🏽

RyanGWU82 replied:
I was able to remove `shaForRef()` but the commit message isn't in the environment or the context object, so I kept that one in here.

kanadgupta commented on src/comments.js:
```diff
@@ -42,24 +42,44 @@ function formatComment(options) {
+  if (owner && repo && prNumber && sha && message) {
+    const commitLink = `https://github.com/${owner}/${repo}/pull/${prNumber}/commits/${sha}`;
+    const commitText = `\`${sha.substring(0, 8)}\` ${message.split('\n')[0]}`;
+    links.push(`:rocket: **Deployed commit:** [${commitText}](${commitLink})`);
```
Ooooh I was wondering why you were constructing the commit links by hand like this... and I just learned that these two links result in different pages!
```
https://github.com/readmeio/heroku-review-app-action/commit/b0814f6230e49037186869dc3a5ceddbf8e47863
https://github.com/readmeio/heroku-review-app-action/pull/4/commits/b0814f6230e49037186869dc3a5ceddbf8e47863
```
Cool with this approach, linking to the commit within the PR is a lot more useful 👍🏽

kanadgupta commented:
Hey @RyanGWU82! Just coming back to this, what are your thoughts on having the Action comment once when the app deploys and have it make subsequent edits to that comment with subsequent redeploys? With the changes in this PR I feel like that'll look a lot cleaner since the user will know which commit has been deployed!

erunion commented on src/comments.js:
```diff
@@ -1,74 +1,157 @@
+  };
+}
+
+let octokit;
```
I'm surprised that this being here isn't throwing any parse or ESLint errors on line 55 where you do `const octokit`.

RyanGWU82 replied:
So am I 🤷🏼  I'll rename this to `memoizedOctokit` (unless you've got a better idea?)

erunion replied:
renaming it should be good enough

RyanGWU82 commented:
@kanadgupta @erunion: This is ready for re-review. Based on the feedback in [today's Slack thread](https://readmeio.slack.com/archives/C5B798RFZ/p1653491855171139), I've updated this to post a single message when the review app is opened, and then append log entries to that when the app is redeployed or shut down. Here's how it looks in action:

<img width="914" alt="Screen Shot 2022-05-25 at 4 02 52 PM" src="https://user-images.githubusercontent.com/313895/170383446-3ea9180f-0358-4a79-8b3d-257b7573e93d.png">

I also fixed the issue where it would show a merge commit message instead of the message from the actual commit in this branch.

erunion reviewed (approved):
:nice:
